discovery/kubernetes/client_metrics.go

```diff
@@ -0,0 +1,265 @@
+}
+func (clientGoRequestMetricAdapter) Observe(verb string, u url.URL, latency time.Duration) {
+	clientGoRequestLatencyMetric.WithLabelValues(u.EscapedPath()).Observe(latency.Seconds())
+}
+
+// Definition of client-go cache metrics provider definition
+type clientGoCacheMetricsProvider struct{}
+
+func (f *clientGoCacheMetricsProvider) Register(registerer prometheus.Registerer) {
+	cache.SetReflectorMetricsProvider(f)
+	registerer.MustRegister(clientGoCacheWatchesDurationMetric)
+	registerer.MustRegister(clientGoCacheWatchesCountMetric)
+	registerer.MustRegister(clientGoCacheListDurationMetric)
+	registerer.MustRegister(clientGoCacheListTotalMetric)
+	registerer.MustRegister(clientGoCacheLastResourceVersionMetric)
+	registerer.MustRegister(clientGoCacheShortWatchesCountMetric)
+	registerer.MustRegister(clientGoCacheItemsInWatchesCountMetric)
+	registerer.MustRegister(clientGoCacheItemsInListCountMetric)
+
+	clientGoCacheWatchesCountMetric.Inc()
+	clientGoCacheListTotalMetric.Inc()
+	clientGoCacheLastResourceVersionMetric.Inc()
+	clientGoCacheShortWatchesCountMetric.Inc()
+}
+func (clientGoCacheMetricsProvider) NewListsMetric(name string) cache.CounterMetric {
+	return clientGoCacheListTotalMetric
+}
+func (clientGoCacheMetricsProvider) NewListDurationMetric(name string) cache.SummaryMetric {
+	return clientGoCacheListDurationMetric
+}
+func (clientGoCacheMetricsProvider) NewItemsInListMetric(name string) cache.SummaryMetric {
+	return clientGoCacheItemsInListCountMetric
+}
+func (clientGoCacheMetricsProvider) NewWatchesMetric(name string) cache.CounterMetric {
+	return clientGoCacheWatchesCountMetric
+}
+func (clientGoCacheMetricsProvider) NewShortWatchesMetric(name string) cache.CounterMetric {
+	return clientGoCacheShortWatchesCountMetric
+}
+func (clientGoCacheMetricsProvider) NewWatchDurationMetric(name string) cache.SummaryMetric {
+	return clientGoCacheWatchesDurationMetric
+}
+func (clientGoCacheMetricsProvider) NewItemsInWatchMetric(name string) cache.SummaryMetric {
+	return clientGoCacheItemsInWatchesCountMetric
+}
+func (clientGoCacheMetricsProvider) NewLastResourceVersionMetric(name string) cache.GaugeMetric {
+	return clientGoCacheLastResourceVersionMetric
+}
+
+// Definition of client-go workqueue metrics provider definition
+type clientGoWorkqueueMetricsProvider struct {
+	logger     log.Logger
+	errMessage string "Error while registering k8s SD workqueue metric."
+}
+
+func (f *clientGoWorkqueueMetricsProvider) Register(registerer prometheus.Registerer) {
+	workqueue.SetProvider(f)
+	registerer.MustRegister(clientGoWorkqueueDepthMetricVec)
+	registerer.MustRegister(clientGoWorkqueueAddsMetricVec)
+	registerer.MustRegister(clientGoWorkqueueLatencyMetricVec)
+	registerer.MustRegister(clientGoWorkqueueWorkDurationMetricVec)
+}
+
+func (f *clientGoWorkqueueMetricsProvider) NewDepthMetric(name string) workqueue.GaugeMetric {
+	metric, err := clientGoWorkqueueDepthMetricVec.GetMetricWithLabelValues(name)
```

Comment:
I didn't want to cause panic due to just minor issue in metric registering but maybe you are right. It would spare me the error handling so for me it's the easier way to go.

I'll stick to the `WithLabelValues` thank you.

Reply:
WithLabelValues is standard.